### ISSUE TITLE
Add terrain/relief view

### ISSUE THREAD

**Issue by ygoe** [mapsme/omim#6960]
I think it would be helpful to have a terrain/relief view like in Google Maps or hiking maps in general, to get an idea of where mountains are on my route. Is that possible? Outdooractive.com has such a view, for example, and their map is also OpenStreetMap-based.

**Comment by clembeb** [comment]
This topographic rendering could be like [OpenTopoMap](https://opentopomap.org/#map=15/48.01856/-2.17529). OpenTopoMap use OSM and SRTM data, it's very usefull for hiking an biking.

**Comment by clembeb** [comment]
The complete map style of OpenTopoMap can be found [here on Github](https://github.com/der-stefan/OpenTopoMap)

**Comment by burivuh** [comment]
Hi!
We will not show terrain in offline mode - this will eat lots of space.
Maybe some day we will do it in online mode - but this is not our first priority.

**Comment by ygoe** [comment]
About how much space are we talking? I was rather surprised to see that all maps of my 5000 km travel route take just around 2 GB. I won't need height data for all of that, just where I'm hiking. So if space is an issue, it could be an optional download per map region.

**Comment by clembeb** [comment]
I don't understand why a new topographic map styles will eat lots of space ? The only new data is the countour lines which could be an optional download par map region like @ygoe say.

**Comment by highbuilder** [comment]
That would be a super option. Perhaps you could offer it to download it seperatly.

**Comment by ferdymercury** [comment]
Vote up

**Comment by lechinoix** [comment]
This is the main reason I am turning to fatmap today instead of mapsme (which is indeed eating A LOT of space offline but worth it). No need for satellite images, just a level line layer would be enough. Not sure to understand why it could not be implemented as you can already see the topographic profile of itinerary.

**Comment by brunetton** [comment]
![](http://bloximages.chicago2.vip.townnews.com/herald-review.com/content/tncms/assets/v3/editorial/1/a5/1a5bb95a-4566-11e2-b4d0-001a4bcf887a/50ca3e9862e4f.preview-620.jpg) 

This feature has been added today 🎉 (I was on the verge to add my 2 cents comment here as I'm also using Fatmap for this feature) and **it have been added today** (Android version 10.1.3) !

Thanks dev team; this is an huge improvement that will change my life (and my mother's and father's lifes, as well as many many other people around the world).

**Comment by keunes** [comment]
I also saw it announced in-app. Can anyone share a screenshot of how this should look? I have the updated app, and new maps. Still, in Brussels I don't see any height-indication, while when cycling this would be useful.

**Comment by brunetton** [comment]
> I also saw it announced in-app. Can anyone share a screenshot of how this should look? I have the updated app, and new maps. Still, in Brussels I don't see any height-indication, while when cycling this would be useful.

Sometimes you have to zoom in to view the curves. Here's a screen shot taken around 45.088527, 2.755106

![Screenshot_20200727_142602_com mapswithme maps pro](https://user-images.githubusercontent.com/86322/88541757-991e1e80-d015-11ea-853b-af90c008b77f.jpg)

**Comment by ygoe** [comment]
That presentation is a bit careless/uninspired (what's the word for it?). Just plotting a few lines. I find it hard to get an idea of the terrain, it's like reading numbers and having to imagine the rest. A bit of a shading like on OpenTopoMap or other comparable maps (even Google Maps terrain layer) would have a great impact.

**Comment by brunetton** [comment]
> That presentation is a bit careless/uninspired (what's the word for it?). Just plotting a few lines. I find it hard to get an idea of the terrain, it's like reading numbers and having to imagine the rest. A bit of a shading like on OpenTopoMap or other comparable maps (even Google Maps terrain layer) would have a great impact.

Of course, and it's very probable that the devs are aware of it, but as the rendering is dynamic (so in real-time) adding a shading algorithm may affect rendering efficiency and global fluidity. And it's (maybe, I don't know) quite hard to implement. 
I'm personally very satisfied with this first iteration, it's waaaayyyy better than nothing. And there will probably be more iterations coming with new options (like I personally would appreciate: relief lines color choose)

**Comment by keunes** [comment]
The lines are hard to see indeed - colours would be helpful.
However, a more fundamental issue, IMHO: the lines only show when zoomed in quite much. This makes the terrain option quite useless when (manually, without planner) navigating: currently zooming in far is required for seeing the lines, while zooming out a bit is very often necessary for navigating.

**Comment by brunetton** [comment]
For the record, someone opened an issue about shading: https://github.com/mapsme/omim/issues/12777
